src/aspect.ts

```diff
@@ -1,27 +1,80 @@
 import { Handle } from './interfaces';
 import { createHandle } from './lang';
 
+/**
+ * An object that provides the necessary APIs to be MapLike
+ */
+export interface MapLike<K, V> {
+	get(key: K): V;
+	set(key: K, value?: V): this;
+}
+
+/**
+ * An internal type guard that determines if an value is MapLike or not
+ *
+ * @param value The value to guard against
+ */
+function isMapLike(value: any): value is MapLike<any, any> {
+	return value && typeof value.get === 'function' && typeof value.set === 'function';
+}
+
+export interface Indexable {
```

Comment:
I believe the exported interfaces should be in alphabetical order as per https://github.com/dojo/meta/blob/master/STYLE.md#ordering